### ISSUE TITLE
Doctrine lifecycle postLoad does not fire, clear() after LoaderInterface::load?

### ISSUE THREAD

**Issue by dkarlovi** [theofidry/AliceDataFixtures#84]
This was quite the challenge to debug so I'm just putting it here, at least for Google to find. It's very simple and perfectly uinderstandable if you know about it.

TLDR If you're persisting fixtures for use in tests, note that **some Doctrine's lifecycle events such as `postLoad` will not fire** if you're testing loading those items (or have custom handlers).

Reason is simple: you're fetching an entity by ID? Entity manager already has it (since the fixture loader just created it), it will NOT:

1. do a query
2. trigger `postLoad` event(s)

If you want the system to behave as it should under test, make sure to `clear()` the entity manager(s) after loading the fixtures.

**Comment by theofidry** [comment]
:+1:, maybe a doc entry about it would be good

**Comment by dkarlovi** [comment]
I'll say. :sob:

**Comment by theofidry** [comment]
Maybe a quick comment in [basic usage](https://github.com/theofidry/AliceDataFixtures#basic-usage) would do the trick, we just need to mention that the entities are in the entity manager so events such as postLoad won't be triggered when loading one of the loaded entity by ID

**Comment by antoniocambados** [comment]
Thank GOD for your bugfix. Tests started to fail in my project all of the sudden, because inversed relations ceased to work :disappointed:

**Comment by dkarlovi** [comment]
@antoniocambados no bugfix for this issue was produced, that I know of?

**Comment by antoniocambados** [comment]
@dkarlovi I meant I solved the problems I was having after calling `clear()`, like you said in this issue.

**Comment by dkarlovi** [comment]
@antoniocambados oh, awesome, glad it helped someone! :+1: 

@theofidry this seems like an actual problem IMO, why not just `clear()` after load instead of adding docs?

**Comment by theofidry** [comment]
If you do so and then change the entities, those entities will be detached
so you would need to retach them before doing anything or you will end up
with duplicate entries in the database

**Comment by dkarlovi** [comment]
@theofidry IMO, that issue would be **much** easier to debug than this one.

**Comment by theofidry** [comment]
Sure but it doesn't make sense in how you use it. Unless you are loading objects for a demo env or something, it means you will need to refresh all the objects you plan to use before using them. So it makes it very inconvenient to use...

**Comment by dkarlovi** [comment]
Maybe registering separate services, one (default) clears after load, the other doesn't? If you need to edit, you use the other one, but by default everything works as you'd (probably, with two samples) expect?

**Comment by theofidry** [comment]
Well the thing is most usages I found clearing up after just breaks everything, so this wouldn't be the default. It could be in HautelookAliceBundle for example where you are mostly using it for loading things in the DB via the command and not manipulate the objects right after, but not if you are creating the objects to manipulate them in your PHPUnit tests.

Also it's easier to do a clear() call rather that refreshing all the entities.

Maybe we could add an optional parameter during the loader though where the default value is configurable like the purge mode,

**Comment by dkarlovi** [comment]
Yes, I'm using it from Behat (for API Platform) and don't modify objects at all (directly, that is), it's just used to get the DB to a known state. When the fixtures are loaded, it should be handed over to my app.

> Maybe we could add an optional parameter during the loader though where the default value is configurable like the purge mode,

Sounds good too. :+1:

**Comment by theofidry** [comment]
>Any reason why a fixture loader should not clear the manager?

I tried to introduce this feature once but had to be reverted because was causing too many issues for people which wanted to re-use those objects. Indeed if you clear it, it means when you are using one of the objects, you have to attach it again manually or it will appear as a new object (and potentially result in a duplicated value conflict)

**Comment by theofidry** [comment]
Reading this thread again: I don't think we can have a nice fix for it as it depends of the use cases, so I would stick to a doc entry.
